hackathon_site/hardware/serializers.py

```diff
@@ -41,10 +41,23 @@ def get_unique_hardware_count(obj: Category) -> int:
 
 class OrderListSerializer(serializers.ModelSerializer):
     hardware_set = HardwareSerializer(many=True, read_only=True)
+    team_code = serializers.SerializerMethodField()
 
     class Meta:
         model = Order
-        fields = ("id", "hardware_set", "team", "status", "created_at", "updated_at")
+        fields = (
```

Comment:
@EdwardJTL and I were talking about this and I forget if we ever wrote it down, but we'll also want to update this view later on so that:
- If the user has order view permissions, then the view behaves like it currently does.
- If the user doesn't have order view permissions, but has a profile, then we automatically filter and return only their team's orders

That way, we can use the same API endpoint for the participant and admin side of the site without having to duplicate a bunch of view logic and (especially) test code.

Reply:
Sounds good! @Leo6Leo's incident API ticket does some of this (it has an OrderItemSerializer) [https://github.com/ieeeuoft/hackathon-template/pull/300/files#diff-b2de03caba10b23014e48bd4882fd1da20367e83dd6cf30df1f14debac0e817bR42](it looks very basic right now)

Reply:
I think you made an issue for what you described above for the permissions https://ieeeuoft.atlassian.net/browse/IEEE-68